src/subscription.ts

```diff
@@ -738,6 +782,25 @@ export class Subscription extends Subscriber {
   snapshot(name: string) {
     return this.pubsub.snapshot.call(this, name);
   }
+  /**
+   * Watches for incoming message event handlers and open/closes the
+   * subscriber as needed.
+   *
+   * @private
+   */
+  private _listen(): void {
+    this.on('newListener', event => {
+      if (!this.isOpen && event === 'message') {
+        this._subscriber.open();
```

Comment:
It depends on the error. If we fail to open a gRPC channel within 5 minutes or so, then we'll emit a deadline error and give up. If we open a channel but all the streaming connections error out with a not retry-able error, we will again give up and emit what the error was.

Otherwise if we detect an open channel we'll continuously create streams so long as the error itself is retry-able.